sdks/java/io/parquet/src/test/java/org/apache/beam/sdk/io/parquet/ParquetIOTest.java

```diff
@@ -126,4 +128,57 @@ public void testReadDisplayData() {
 
     Assert.assertThat(displayData, hasDisplayItem("filePattern", "foo.parquet"));
   }
+
+  public static class TestRecord {
+    String name;
+
+    public TestRecord(String name) {
+      this.name = name;
+    }
+  }
+
+  @Test(expected = org.apache.beam.sdk.Pipeline.PipelineExecutionException.class)
```

Comment:
Does it fail only in case if we obtain schema using Avro `ReflectData`?

Reply:
For schemas that include a namespace and name that map back to a valid class in the classpath, the AvroRecordConverter.start() method will attempt to create an instance of the given class when the SpecificData model is in use (this doesn't happen when the name/namespace aren't set). So yes - when the schema is obtained with ReflectData, the name and namespace would be 'TestRecord' and 'beam.sdks.java.io.parquet.test', and since this class doesn't implement a no-arg constructor and the other methods expected by SpecificData the runner ends up throwing a NoSuchMethodException.